client_app/src/components/shelter/ShelterDashboard.js

```diff
@@ -3,8 +3,64 @@ import OpenRequest from "./OpenRequests";
 import PastVolunteersContainer from "./PastVolunteersContainer";
 import ShiftContainer from "./ShiftContainer";
 import "../../styles/index.css";
+import { useState, useRef, useEffect } from "react";
+import setHours from "date-fns/setHours";
+import setMinutes from "date-fns/setMinutes";
+import setSeconds from "date-fns/setSeconds";
+import setMilliseconds from "date-fns/setMilliseconds";
+import { SERVER } from "../../config";
 
 function ShelterDashboard() {
+  const [shiftDetails, setShiftDetails] = useState([]);
+  let shelterId = 30207;
+  let startTime = 
+      setHours(
+        setMinutes(setSeconds(setMilliseconds(new Date(), 0), 0), 0),
+        new Date().getHours() + 1,
+      );
+
+  const initialized = useRef(false);
+  
+  useEffect(() => {
+    let start = setMilliseconds(setSeconds(setMinutes(setHours(startTime, 0), 0), 0), 0);
+    const oneHourInMs = 3600 * 1000;
+    const validResponses = [];
+
+    if (!initialized.current && shelterId) {
+      initialized.current = true;
+      const fetchPromises = [];
+      for (let i = 0; i < 24; i++) {
+        const intervalStartEpoch = start.getTime() + (i * oneHourInMs);
+        const intervalEndEpoch = intervalStartEpoch + oneHourInMs;
+        let request_endpoint = `${SERVER}/getvolunteers/${shelterId}?filter_start_after=${intervalStartEpoch}&filter_end_before=${intervalEndEpoch}`;
+        fetchPromises.push(
+          fetch(request_endpoint, {
+            method: "GET",
+            headers: {
+              "Content-Type": "application/json",
+              Authorization: "volunteer@slu.edu",
+            },
+          })
+          .then((response) => response.json())
+          .then((data) => {
+            if (Array.isArray(data) && data.length > 0) {
```

Comment:
just in case, add check if data doesnt exist or is null

Reply:
In this if statement, it checks two conditions at the same time: it needs to be an array, and its length is greater than 0, which means it cannot be empty.